src/components/Header/Search/Search.component.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import {
+  Container,
+  SearchIcon,
+  BackIcon,
+  IconContainer,
+  SearchIconContainer,
+  SearchBarContainer,
+} from './Search.styles';
+import { useAuth } from '../../../providers/Auth';
+import { storage } from '../../../utils/storage';
+
+const Search = () => {
+  const { state, setState } = useAuth();
+  const localVideoStoraged = storage.get('localVideoStoraged');
+  const handleSearchBarState = () =>
+    setState({
+      ...state,
+      searchbar: (prevState) => !prevState,
+    });
+
+  const handleSearchString = (event) => {
+    const { value } = event.target;
+    setState({
+      ...state,
+      searchString: value,
+    });
+  };
+
+  const handleSearch = () => {
+    if (localVideoStoraged.searchString !== state.searchString) {
+      setState({
+        ...state,
+        isLoading: true,
+      });
+    }
+  };
+
+  return (
+    <>
+      <Container searchbar={state.searchbar}>
+        <IconContainer onClick={handleSearchBarState}>
+          <BackIcon />
+        </IconContainer>
+        <SearchBarContainer>
+          <input
+            placeholder="Search"
+            onChange={handleSearchString}
```

Comment:
I have noticed a possible bug here, I try to do a search for another word, like "google", and the results are shown, but when I try to do a new search again, I can do it, the app freezes. Also, it would be great to trigger the search with the "Enter" key